services/storage/src/simcore_service_storage/dsm.py

```diff
@@ -491,15 +491,57 @@ async def try_update_database_from_storage(
     )
     async def auto_update_database_from_storage_task(
         self, file_uuid: str, bucket_name: str, object_name: str
-    ):
+    ) -> Optional[FileMetaDataEx]:
         return await self.try_update_database_from_storage(
             file_uuid, bucket_name, object_name, silence_exception=True
         )
 
-    async def upload_link(self, user_id: str, file_uuid: str, as_presigned_link: bool):
+    async def update_metadata(
+        self, file_uuid: str, user_id: int
+    ) -> Optional[FileMetaDataEx]:
+        async with self.engine.acquire() as conn:
+            can: Optional[AccessRights] = await get_file_access_rights(
+                conn, int(user_id), file_uuid
+            )
+            if not can.write:
+                message = f"User {user_id} was not allowed to upload file {file_uuid}"
+                logger.debug(message)
+                raise web.HTTPForbidden(reason=message)
+
+        bucket_name = self.simcore_bucket_name
+        object_name = file_uuid
+        return await self.auto_update_database_from_storage_task(
+            file_uuid=file_uuid,
+            bucket_name=bucket_name,
+            object_name=object_name,
+        )
+
+    async def delete_metadata(self, user_id: int, file_uuid: str) -> None:
```

Comment:
How is this in sync with ``delete_file``? I mean, if you have a handler to delete metadata but it still exists in s3 .. how this inconsistent state gets resolved?

Reply:
I agree with @pcrespov here. What about having the handler delete_file handle it as well? from a client perspective it would be clearer, cause using delete_metadata implies that the client code knows the internals of storage.